src/redux/Rockets/rocketSlice.js

```diff
@@ -24,10 +24,20 @@ export const rocketSlice = createSlice({
         loading: action.payload,
       };
     },
+
+    addReservation(state, action) {
+      state.rocketsArray.map((rocket) => {
+        const mappedRocket = rocket;
+        if (mappedRocket.id === action.payload) {
+          mappedRocket.reserved = true;
+        }
+        return mappedRocket;
+      });
+    },
   },
 });
 
-export const { setLoading, initRocketsData } = rocketSlice.actions;
+export const { setLoading, initRocketsData, addReservation } = rocketSlice.actions;
 
 export const fetchRocketsData = createAsyncThunk(FETCH_ROCKETS_DATA,
```

Comment:
[OPTIONAL] To optimize the full potentials of @redux/toolkit, i think you can use extrareducers to keep track of all the stages of the async actions(pending, fulfilled and rejected).

Reply:
I have an idea of `extraReducers`. For now, I am just trying to keep the logic as simple as possible. I may use it after completing all the basic requirements.